app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,10 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      order.line_items.each do |line_item|
+        line_item.update_column(:pre_tax_amount, line_item.discounted_amount)
```

Comment:
@athal7 this is the correct place to fix it (barring deeper refactoring in solidus itself). [It's the tax rate that sets `pre_tax_amount`](https://github.com/solidusio/solidus/blob/4aa802a/core/app/models/spree/tax_rate.rb#L74) and solidus_avatax overrides how Solidus interacts with tax rates (happens in this method) so this seems like a good place to do it.